ocaml/xapi/xapi_cluster_host.ml

```diff
@@ -349,6 +353,14 @@ let enable ~__context ~self =
         else
           Cluster_interface.(IPv4 (ipstr_of_address ip_addr))
       in
+      let cluster_ref = Db.Cluster_host.get_cluster ~__context ~self in
+      let cluster_stack =
```

Comment:
Splitting the `stack` and `version` seems annoying to me.
Given the following has been defined:
```
type t = Corosync2 | Corosync3
```
How about encoding the version into the type values? E.g.
```
module Cluster_stack = struct
  let of_string = function
    | "corosync" -> Corosync2
    | "corosync3" -> Corosync3
    | _ -> ...
end
```